spec/e2e/relationship/persistence/query_factory_spec.rb

```diff
@@ -103,6 +103,30 @@ def self.count
         expect(from_node.reload.to_classes).to be_empty
       end
 
+      context 'concurrent update' do
+        before do
+          allow_any_instance_of(ActiveGraph::Node::Query::QueryProxy).to receive(:replace_with).and_wrap_original do |original, *args|
+            $concurrency_queue << 'ready'
+            Thread.stop
```

Comment:
I think the right thing would be to stop at `delete_rels_for_nodes` after the rels are read so you can force failure with just 2 threads not 100. And then after adding lock see it passing.